src/routes/courses.js

```diff
@@ -127,8 +127,12 @@ router.post("/ids", async (req, res) => {
       where: conditions,
     });
     if (courses) {
+      const courseObjects = course_post_process(courses);
+      const coursesSortByIds = ids.map((id) =>
+        courseObjects.find((course) => course.id === id)
+      );
       res.status(200).send({
-        courses: course_post_process(courses),
+        courses: coursesSortByIds,

```

Comment:
Bug 2: return courses w/ wrong order

Reply:
You can sort it in a faster way. `O(n^2)` -> `O(nlogn)`
I will fix it.
```javascript
itemsArray.sort(function(a, b){  
  return sortingArr.indexOf(a) - sortingArr.indexOf(b);
});
```